pkg/agent/core/state.go

```diff
@@ -88,83 +88,86 @@ type LogConfig struct {
 // State holds all of the necessary internal state for a VM in order to make scaling
 // decisions
 type State struct {
-	// ANY CHANGED FIELDS MUST BE UPDATED IN dumpstate.go AS WELL
+	internal internalState
+}
 
-	config Config
+// one level of indirection below State so that the fields can be public, and JSON-serializable
+type internalState struct {
```

Comment:
This is more sort of a personal opinion, but you may want to name this `state` instead. `internalState` is kinda redundant, since `state` implies that it's internal because it's unexported.

Reply:
Hm, yeah, that makes sense. I would _kind of_ find having two "state"s confusing, but perhaps it's worth it. I'll make the change.

Reply:
aab8174a4cc76a7b17b0c236d90866e47998fb55 - overwritten by later rebase.